File: src/breslin/clientside/application/states/applicationMain.cpp
//header
#include "applicationMain.h"


//log
#include "../../tdreamsock/dreamSockLog.h"

//application
#include "../applicationBreslin.h"

//game
#include "../../game/gamePartido.h"

//state machine
#include "../../../statemachine/stateMachine.h"

/***************************************
*	CONSTRUCTORS
***************************************/
ApplicationMain::ApplicationMain(ApplicationBreslin* application)
{
	mApplication = application;
}

ApplicationMain::~ApplicationMain()
{

}

void ApplicationMain::enter()
{
	mApplication->createMainScreen();
	mApplication->showMainScreen();
}

void ApplicationMain::execute()
{
	if (mApplication->mButtonHit == mApplication->mButtonJoinGameA)
	{
		mApplication->mButtonHit = NULL;
 		mApplication->setGame(new Game(mApplication));
		mApplication->getGame()->createStates();
		mApplication->getGame()->setStates();
		mApplication->sendJoinGame(1);
		mApplication->mStateMachine->changeState(mApplication->mApplicationPlay);
	}

	if (mApplication->mButtonHit == mApplication->mButtonLogout)
	{
		mApplication->mButtonHit = NULL;
		mApplication->sendLogout();
	}
	
	if (mApplication->mLoggedIn == false)
	{
		mApplication->mStateMachine->changeState(mApplication->mApplicationLogin);
	}

	if (mApplication->mButtonHit == mApplication->mButtonExit)
	{
		mApplication->sendLogout();
		mApplication->mStateMachine->changeState(NULL);
		mApplication->mStateMachine->setGlobalState(NULL);
		mApplication->mButtonHit = NULL;
		mApplication->shutdown();
		mApplication->mShutDown = true;
		delete mApplication;
	}

	if (mApplication->mButtonHit == mApplication->mButtonShutdownServer)
	{
		mApplication->sendShutdownServer();
		mApplication->mButtonHit = NULL;
	}
}

void ApplicationMain::exit()
{
	mApplication->hideMainScreen();
}

File: src/breslin/clientside/application/states/application_main.php
var ApplicationMain = new Class(
{
	
initialize: function(applicationBreslin)
{
	this.mApplicationBreslin = applicationBreslin;
},

enter: function()
{
        this.mApplicationBreslin.createMainScreen();
        this.mApplicationBreslin.showMainScreen();
},

execute: function()
{
	//join game A
	if (this.mApplicationBreslin.mButtonHit == this.mApplicationBreslin.mButtonJoinGameA)
	{
	        this.mApplicationBreslin.mButtonHit = 0;
		this.mApplicationBreslin.setGame(new Game(this.mApplicationBreslin));
		this.mApplicationBreslin.getGame().createStates();
		this.mApplicationBreslin.getGame().setStates();
 		this.mApplicationBreslin.sendJoinGame('1');
                this.mApplicationBreslin.mStateMachine.changeState(this.mApplicationBreslin.mApplicationPlay);
	}

	//logout
	if (this.mApplicationBreslin.mButtonHit == this.mApplicationBreslin.mButtonLogout)
	{
	        this.mApplicationBreslin.mButtonHit = 0;
                this.mApplicationBreslin.sendLogout();
	}

	if (this.mApplicationBreslin.mLoggedIn == false)
	{
                this.mApplicationBreslin.mStateMachine.changeState(this.mApplicationBreslin.mApplicationLogin);
	}

	//exit
	if (this.mApplicationBreslin.mButtonHit == this.mApplicationBreslin.mButtonExit)
	{
                this.mApplicationBreslin.sendLogout();
	        this.mApplicationBreslin.mButtonHit = 0;
	}
},

exit: function()
{
        this.mApplicationBreslin.hideMainScreen();
}

});
